client/landing/stepper/hooks/use-premium-global-styles.ts

```diff
@@ -0,0 +1,23 @@
+import { isEnabled } from '@automattic/calypso-config';
+import { WPCOM_FEATURES_GLOBAL_STYLES } from '@automattic/calypso-products';
+import { useSelect } from '@wordpress/data';
+import { useSite } from 'calypso/landing/stepper/hooks/use-site';
+import { SITE_STORE } from 'calypso/landing/stepper/stores';
+
+export function usePremiumGlobalStyles() {
+	const siteId = useSite()?.ID;
+
+	const hasGlobalStyles = useSelect( ( select ) =>
+		select( SITE_STORE ).siteHasFeature( siteId, WPCOM_FEATURES_GLOBAL_STYLES )
+	);
+
+	// Do not limit Global Styles on sites created before we made it a paid feature. This cutoff
+	// blog ID needs to be updated as part of the public launch.
+	const shouldLimitGlobalStyles = siteId
+		? siteId > 210494207 && ! hasGlobalStyles && isEnabled( 'limit-global-styles' )
+		: isEnabled( 'limit-global-styles' );
+
+	return {
+		shouldLimitGlobalStyles,
+	};
+}
```

Comment:
I wonder if `/landing/stepper` is the right place for this hook, but at the same time, I don't see a more generic location that would be a good fit for it. 🤔 

(Not a blocker, just a consideration.)

Reply:
Yeah, I wondered about that too and [initially added it to the `calypso/lib` path](https://github.com/Automattic/wp-calypso/pull/69082/commits/2c96819a351a473aa7f9b8c675ae9639cdb73df3), but it felt a bit off so I finally moved it to where most of these generic hooks are.